src/HDF5.jl

```diff
@@ -2007,18 +2010,41 @@ function hdf5_to_julia_eltype(objtype)
         T = HDF5Vlen{hdf5_to_julia_eltype(HDF5Datatype(super_id))}
     elseif class_id == H5T_COMPOUND
         N = Int(h5t_get_nmembers(objtype.id))
-        # check if should be interpreted as complex
-        if COMPLEX_SUPPORT[] && N == 2
-          membernames = ntuple(N) do i
-            h5t_get_member_name(objtype.id, i-1)
-          end
-          membertypes = ntuple(N) do i
-            hdf5_to_julia_eltype(HDF5Datatype(h5t_get_member_type(objtype.id, i-1)))
+
+        membernames = ntuple(N) do i
+          h5t_get_member_name(objtype.id, i-1)
+        end
+
+        membertypes = ntuple(N) do i
+          dtype = HDF5Datatype(h5t_get_member_type(objtype.id, i-1))
+          ci = h5t_get_class(dtype.id)
+
+          if ci == H5T_STRING
+            if h5t_is_variable_str(dtype.id)
+              return Cstring
+            else
+              n = h5t_get_size(dtype.id)
+              return FixedString{Int(n)}
```

Comment:
Again I'm guessing it's fine not do `Int` here as well. Unless you foresee negative consequences.

Reply:
This one is actually necessary since NTuple requires the first type parameter to be `Int64`.